common/src/Model/GroupNode.cpp

```diff
@@ -72,53 +73,52 @@ namespace TrenchBroom {
                         group.transform(transformation);
-                        origNodeAndTransformedContents[i] = {groupNode, std::move(group)};
+                        return std::make_pair(nodeToTransform, NodeContents{std::move(group)});
                     },
-                    [&](const EntityNode* entityNode) {
+                    [&](const EntityNode* entityNode) -> TransformResult {
                         auto entity = entityNode->entity();
                         entity.transform(transformation);
-                        origNodeAndTransformedContents[i] = {entityNode, std::move(entity)};
+                        return std::make_pair(nodeToTransform, NodeContents{std::move(entity)});
                     },
-                    [&](const BrushNode* brushNode) {
+                    [&](const BrushNode* brushNode) -> TransformResult {
                         auto brush = brushNode->brush();
-                        brush.transform(worldBounds, transformation, true)
-                            .and_then([&]() {
-                                origNodeAndTransformedContents[i] = {brushNode, std::move(brush)};
-                            })
-                            .handle_errors([&](BrushError){
-                                transformFailed = true;
+                        return brush.transform(worldBounds, transformation, true)
+                            .and_then([&]() -> TransformResult {
+                                return std::make_pair(nodeToTransform, NodeContents{std::move(brush)});
                             });
                     },
-                    [&](const PatchNode* patchNode) {
+                    [&](const PatchNode* patchNode) -> TransformResult {
                         auto patch = patchNode->patch();
                         patch.transform(transformation);
-                        origNodeAndTransformedContents[i] = {patchNode, std::move(patch)};
+                        return std::make_pair(nodeToTransform, NodeContents{std::move(patch)});
                     }
                 ));
             });
 
+            bool transformFailed = false;
+            auto origNodeAndTransformedContents = kdl::collect_values(transformResults, kdl::overload(
+                [&](const auto&) {
+                    transformFailed = true;
+                }
+            ));
+
             if (transformFailed) {
                 return UpdateLinkedGroupsError::TransformFailed;
             }
 
             // Move into map for easier lookup
-            auto resultsMap = std::unordered_map<const Node*, NodeContentType>{};
+            auto resultsMap = std::unordered_map<const Node*, NodeContents>{};
             for (auto& [origNode, transformedContents] : origNodeAndTransformedContents) {
-                resultsMap[origNode] = std::move(transformedContents);
+                resultsMap.emplace(origNode, std::move(transformedContents));
```

Comment:
Why change this? `[origNode] = ` is nicer syntax..

Reply:
Reading a bit I guess you need emplace if the map value type isn't default constructible.

Reply:
Yes, exactly, emplace is also faster because you safe one creation, but I guess that won't be noticable anyway.